torch/csrc/jit/argument_spec.h

```diff
@@ -47,9 +53,15 @@ struct ArgumentInfo {
   }
 
  private:
+  unsigned is_nontensor_optional_ : 1;
```

Comment:
The way this is encoded is pretty weird. It would be clearer as something like:

// Argspec has:

std::vector<ArgumentInfo> tensor_args; // tensor information recorded
std::vector<bool> optional_presence; // for each optional type true if present, false if not

```
// Related Argspec commands are:
SPECIALIZE_OPTIONAL_TENSOR, // push present/not-present to optional_presense, and if present, add the tensor information to tensor_args
SPECIALIZE_TENSOR, // push tensor information to tensor_args
SPECIALIZE_OPTIONAL, // push present/not-present to optional_presense, 
```

Reply:
So elegant! Thanks!